rc/extra/racer.kak

```diff
@@ -45,7 +72,7 @@ define-command racer-complete -docstring "Complete the current selection with ra
                 }'
             )
             printf %s\\n "evaluate-commands -client '${kak_client}' %{
-                set-option buffer=${kak_bufname} racer_completions %@${compl}@
+                set-option buffer=${kak_bufname} racer_completions %@${compl: : -1}@
```

Comment:
This doesn't work on MacOS. Could we do POSIX style, like `${compl%?}` ?

Reply:
Yeah, sorry I did not notice that during review, `${blah:<pos>:<len>}` is a bash extension, seems here you want to remove the last character, so `${compl%?}` should do the same portably. I dont have racer setup here, so I cannot test. Could you validate that this works ?

Reply:
If the last character is unknown, you could still use the following:

```
expr "${compl}" : '\(..*\).$'
```

Reply:
I don't have much bash experience (I had to google for this solution), so...

@mawww I've tried the `${compl%?}` approach and seems to work on linux (I haven't been able to test on this on a mac).

Question for you all: the problem is that awk adds an extra `:` (the row separator) at the end, do you think something more specific that ensures that what's removed is actually a `:` is needed?

Reply:
I tested `${compl%?}` in on Mac, it does work there.

Reply:
I don't speak awk, but I guess this line `gsub(/:/, "\\:", candidate)` (https://github.com/marcesquerra/kakoune/blob/f6a3fd2e4ed8857ffb1900fbb17370eaab277500/rc/extra/racer.kak#L70) adds `:` after every candidate, therefore we could be pretty sure that `:` is always the last character, except of empty candidates list case.